aksel.nav.no/website/scripts/update-examples/parts/filter-code.ts

```diff
@@ -1,23 +1,164 @@
+    ) {
+      return;
+    }
+
+    // Read file and parse code
+    const importedCode = fs.readFileSync(
+      path.resolve(path.dirname(filePath), `${importPath}.tsx`),
+      "utf-8",
+    );
+    const parsedImportedCode = j(importedCode);
+
+    // Imports from the __parts-inline directory are components that should be inlined in the example JSX.
+    if (importPath.includes("/__parts-inline/")) {
+      // Extract the JSX from the component.
+      // We assume it's written as an arrow function with implicit return.
+      // This code will only extract the first arrow function in the file. Other code is ignored.
+      const arrowFunctions = parsedImportedCode.find(j.ArrowFunctionExpression);
+      const arrowFunctionCode = j(arrowFunctions.nodes()[0].body)
+        .toSource()
+        .slice(1, -1); // Remove leading and trailing parenthesis
+      const identifier = importDecl.value.specifiers[0].local.name; // The part between "import" and "from"
+      if (!identifier) return;
+      jsxToReplace[identifier] = arrowFunctionCode;
+    }
+    // Imports from the __parts directory are variables/functions that should be injected bellow the example JSX.
+    else if (importPath.includes("/__parts/")) {
+      // For simplicity, we don't filter out unused code.
+      const cleanedCode = importedCode
+        .replace(/import [^;]+;/g, "")
+        .replace(/export [^;]+;/g, "")
+        .trim();
+      codeToInject.push(cleanedCode);
+    }
+
+    // Copy imports from the imported file to the example code
+    handleDependencies(importDeclarations, importDecl, parsedImportedCode);
+
+    // Remove the import declaration for the inlined component
+    importDecl.prune();
+  });
+
+  if (!Object.keys(jsxToReplace).length && !codeToInject.length) {
+    return code;
+  }
+  code = parsedCode.toSource({ lineTerminator: "\n" });
+
+  // Inject code
+  Object.entries(jsxToReplace).forEach(([identifier, arrowFunctionCode]) => {
+    const regex = new RegExp(`<${identifier} [^/]*/>`, "g");
+    code = code.replace(regex, arrowFunctionCode);
+  });
+  code += "\n\n" + codeToInject.join("\n\n");
+
+  code = code.replace(/\n\nimport /g, "\nimport "); // prune() adds empty line between imports
+
+  return await prettier.format(code, { parser: "typescript" });
+}
+
+/**
+ * Copies imports from the imported file to the example code.
+ * Only handles aksel/ds and react package imports. (So no recursiveness.)
+ */
+function handleDependencies(
+  importDeclarations: Collection<namedTypes.ImportDeclaration>,
+  currentImportDecl: ASTPath<namedTypes.ImportDeclaration>,
+  parsedImportedCode: Collection<any>,
+) {
+  parsedImportedCode.find(j.ImportDeclaration).forEach((nestedImportDecl) => {
+    const packageName = nestedImportDecl.value.source.value?.toString() || "";
+    if (
+      (!packageName.includes("@navikt/") && packageName !== "react") ||
```

Comment:
If i'm not mistaken this code is also ran for all the examples? If so we need to make a quick check and fix some of the stories. Some of the datepicker/monthopicker uses `date-fns` as an import last time i checked.

Reply:
This particular code only runs on files in __parts and __parts-inline.